lib/zendesk_apps_support/assets/installed.js.erb

```diff
@@ -1,6 +1,28 @@
 (function() {
+  var self = this;
+  var reporter = window.ZendeskReporter ? new window.ZendeskReporter() : {};
+  function wrapped$() {
+    reporter.increment && reporter.increment(
+      'app_framework.app_scope.violated',
+      1,
+      ['origin:' + self.location.hostname]
+    );
+    return self.$.apply(null, arguments);
+  }
+
+  var wrappedWindow = this.$.extend(true, {}, self, { $: wrapped$ });
+  wrappedWindow.window = wrappedWindow;
+  wrappedWindow.top = wrappedWindow;
+
   <% appsjs.each do |appjs| %>
-    <%= appjs %>
+    <% if apps_zaf_naughty_v1_logging %>
+      <%= appjs.gsub(
+        /}\(\)\)\;*\s*;*\s*}\s*var\s*app\s*=\s*ZendeskApps\.defineApp\(source\)/mx,
+        "}.bind(wrappedWindow)());\n;\n}\n\nvar app = ZendeskApps.defineApp(source)"
```

Comment:
I wonder if we could format this properly and convert the whitespace and line breaks programatically to make it easier to read?

Reply:
I'm not sure how to do that. In the replaced text, we can just remove the whitespaces, if that will help with readability.

Reply:
Or I can add a comment block that tell you what is the regex replacing and with what in a nice formatted way.

Reply:
A comment here would be good 👍